pyproject.toml

```
@@ -125,16 +119,14 @@ exclude_also = [
 
 [tool.pytest.ini_options]
 addopts = [
-    # "--import-mode=importlib",  # TODO: enable: https://github.com/scverse/anndata/issues/1451
+    "--import-mode=importlib",
     "--strict-markers",
     "--doctest-modules",
     "--pyargs",
 ]
 filterwarnings = [
-    'ignore:Support for Awkward Arrays is currently experimental',
-    'ignore:Outer joins on awkward\.Arrays',
-    # TODO: replace both lines above with this one once we figured out how prevent ImportPathMismatchError
-    # 'ignore::anndata._warnings.ExperimentalFeatureWarning',
+    'ignore::anndata._warnings.ExperimentalFeatureWarning',
+    'ignore::anndata.OldFormatWarning',
```

```suggestion
    'ignore:Support for Awkward Arrays is currently experimental',
    'ignore:Outer joins on awkward\.Arrays',
```

This seems to clear up issues with pytest-cov, and still lets us do coverage in parallel

---

That’s not right, you just deleted the `OldFormatWarning` line without replacement.

Unless the `warnings.filterwarnings` line in conftest.py was redundant.

Also if we go back to hacky message matching, we need to link the two issues that block this:

- https://github.com/nedbat/coveragepy/issues/1696
- https://github.com/pytest-dev/pytest-cov/issues/627

we need either of them to be fixed to have a satisfying solution (either keep using `pytest --cov` if that fix comes first or switch to `PYTHONSAFEPATH=1 coverage run -m pytest` if that fix comes first)

---

Wait, why would we switch to `PYTHONSAFEPATH=1 coverage run -m pytest`?

I would really like to keep pytest-cov going so that I can collect coverage locally in ~30 seconds instead of ~5 min.

---

Yeah, seems the alternatives don’t work as reliably: https://github.com/nedbat/coveragepy/issues/1341